test/parallel/test-worker-name.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const common = require('../common');
+common.skipIfInspectorDisabled();
+const assert = require('assert');
+const { Worker, isMainThread } = require('worker_threads');
+
+const { Session } = require('inspector');
+
+if (isMainThread) {
+  const name = 'Hello Thread ';
+  new Worker(__filename, {
+    workerData: 'Test Worker',
+    name,
+  });
+
+  const session = new Session();
+  session.connect();
+  session.on('NodeWorker.attachedToWorker', ({ params: { workerInfo } }) => {
+    const id = workerInfo.id;
+    const expectedTitle = `[Worker ${id}] ${name}`;
```

Comment:
Ok it seems, the test wasn't quite right this typo here should have cause the test to fail but it didn't and now i am confused what to do, is the inspector session not starting on the main thread?

Reply:
ok it seems directly running the test as `out/Release/node test/parallel/test-worker-name.js` the test fails whereas doing `python3 tools/test.py test/parallel/test-worker-name.js` the test doesnt fail which is quite confusing

any ideas? @addaleax ?

Reply:
Does this callback get fired when you run it with the python test runner? I think we should surround the callback with a `common.mustCall()` to verify.

Reply:
ok got it surrounded by `common.mustCall()` still no effect

```console
debadreechatterjee@Debadree-MacBook-Pro node % python3 tools/test.py test/parallel/test-worker-name.js
[00:00|% 100|+   1|-   0]: Done                    

All tests passed.
debadreechatterjee@Debadree-MacBook-Pro node % out/Release/node test/parallel/test-worker-name.js
(node:38708) [ERR_ASSERTION] AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ '[worker 1] Hello Thread'
- '[Worker 1] Hello Thread'
(Use `node --trace-warnings ...` to show where the warning was created)
```

Reply:
Oh, actually, the test doesn't really work when you run it with `./node` either. It's still exiting with a 0 exit code, so I think you would need to find a way to forward the `id` and the `title` to some other place outside this callback where you could run the assertions.